src/Mod/Material/Gui/MaterialsEditor.cpp

```diff
@@ -886,54 +887,120 @@ void MaterialsEditor::refreshMaterialTree()
     fillMaterialTree();
 }
 
-void MaterialsEditor::updatePreview() const
+bool MaterialsEditor::updateTexturePreview() const
 {
-    if (_material->hasAppearanceProperty(QString::fromStdString("AmbientColor"))) {
-        QString color = _material->getAppearanceValueString(QString::fromStdString("AmbientColor"));
+    bool hasImage = false;
+    QImage image;
+    double scaling = 99.0;
```

Comment:
Why define this up here? And why default to 99?

Reply:
That was a testing thing. I'll correct, but it does no harm as it isn't used (yet)